apps/blockchain/lib/blockchain/interface/account_interface.ex

```diff
@@ -37,9 +40,144 @@ defmodule Blockchain.Interface.AccountInterface do
     }
   end
 
-  @spec commit_storage(t()) :: EVM.state()
+  @spec commit(t()) :: t()
+  def commit(account_interface) do
+    account_interface.cache
+    |> Cache.commit(account_interface.state)
+    |> new()
+  end
+
+  @spec commit_storage(t()) :: t()
   def commit_storage(account_interface) do
-    Cache.commit(account_interface.cache, account_interface.state)
+    new_state = Cache.commit_storage(account_interface.cache, account_interface.state)
+
+    new(new_state, %Cache{accounts_cache: account_interface.cache.accounts_cache})
```

Comment:
Should we have functions in `Cache` that allow us to create/update the accounts_cache? That way, the `Cache` module is solely responsible for manipulating the data. What do you think?

Reply:
@germsvel This method was unused. I remove it. Cache is committed only with `AccountInterface.commit/1` which delegates to `Cache.commit`